upstairs/src/impacted_blocks.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright 2022 Oxide Computer Company
+
+use super::*;
+
+use bitvec::prelude::*;
+use itertools::Itertools;
+use std::ops::BitAnd;
+
+/// Store a list of impacted blocks for later job dependency calculation
+#[derive(Debug, Default, PartialEq)]
+pub struct ImpactedBlocks {
+    ddef: RegionDefinition,
+
+    // extent id -> list of block offsets
+    blocks: HashMap<u64, BitVec>,
```

Comment:
For the dependency map, this is a bitmap of blocks in the extent that are part of this IO?